core/src/main/java/org/mule/api/processor/DefaultMessageProcessorPathElement.java

```diff
@@ -41,7 +43,7 @@ public void setParent(MessageProcessorPathElement parent)
     @Override
     public List<MessageProcessorPathElement> getChildren()
     {
-        return Collections.unmodifiableList(children);
+        return ImmutableList.copyOf(children);
```

Comment:
Import the static method (though it was not used before with Collection.copy it should be imported like this according to the style rules).

Reply:
Though on second thought, it is appropriate to leave the ImmutableList so that it is clear that it is an immutable list. Sorry.